src/server.jsx

```diff
@@ -2,30 +2,44 @@ import React from 'react';
 import ReactDOMServer from 'react-dom/server';
 import { Provider } from 'react-redux';
 import { StaticRouter as Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import axios from 'axios';
 
 import configureStore from 'configureStore.js';
 import AppView from 'views/AppView.jsx';
+import Html from 'components/Html/Html.jsx';
 
-export default function (props) {
-  const store = configureStore();
+const authApi = req => (
+  axios.get('http://localhost:3000/api/user', {
+    headers: {
+      Cookie: req.headers.cookie,
+    },
+  })
+    .then(r => r.data)
+    .catch(() => null)
+);
+
+export default async function ({ req, res, context }) {
+  const history = createMemoryHistory({
+    initialEntries: [req.url],
+  });
+  const user = await authApi(req);
+  const store = configureStore(history, { user });
```

Comment:
We are hydrating the store with the user data and with the current url, so the routerMiddleWare is hydrated